### PR TITLE
Various tweaks to diagnostic output

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

File: src/librustc_metadata/creader.rs [hunk 1]
```diff
@@ -698,7 +698,9 @@ impl<'a> CrateLoader<'a> {
         let has_global_allocator = match &*global_allocator_spans(krate) {
             [span1, span2, ..] => {
                 self.sess.struct_span_err(*span2, "cannot define multiple global allocators")
-                         .span_note(*span1, "the previous global allocator is defined here").emit();
+                    .span_label(*span2, "can't define a new global allocator")
```
Comment (Centril) [inline src/librustc_metadata/creader.rs]:
```suggestion
                    .span_label(*span2, "cannot define a new global allocator")
```

Comment (estebank) [reply src/librustc_metadata/creader.rs]: done

**Pull request merged**